entitybroker/core-providers/src/java/org/sakaiproject/entitybroker/providers/HomeEntityProvider.java

```diff
@@ -0,0 +1,116 @@
+package org.sakaiproject.entitybroker.providers;
+
+import lombok.Setter;
+
+import org.sakaiproject.content.api.ContentHostingService;
+import org.sakaiproject.content.api.ContentResource;
+import org.sakaiproject.entitybroker.DeveloperHelperService;
+import org.sakaiproject.entitybroker.EntityView;
+import org.sakaiproject.entitybroker.entityprovider.EntityProvider;
+import org.sakaiproject.entitybroker.entityprovider.annotations.EntityCustomAction;
+import org.sakaiproject.entitybroker.entityprovider.capabilities.ActionsExecutable;
+import org.sakaiproject.entitybroker.entityprovider.capabilities.AutoRegisterEntityProvider;
+import org.sakaiproject.entitybroker.entityprovider.capabilities.Describeable;
+import org.sakaiproject.entitybroker.entityprovider.capabilities.Outputable;
+import org.sakaiproject.entitybroker.entityprovider.capabilities.Sampleable;
+import org.sakaiproject.entitybroker.entityprovider.extension.Formats;
+import org.sakaiproject.entitybroker.providers.model.EntityOverview;
+import org.sakaiproject.entitybroker.util.AbstractEntityProvider;
+import org.sakaiproject.exception.IdUnusedException;
+import org.sakaiproject.exception.PermissionException;
+import org.sakaiproject.exception.ServerOverloadException;
+import org.sakaiproject.exception.TypeException;
+import org.sakaiproject.site.api.Site;
+import org.sakaiproject.site.api.SiteService;
+import org.sakaiproject.tool.api.SessionManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+public class HomeEntityProvider extends AbstractEntityProvider implements AutoRegisterEntityProvider, Describeable, EntityProvider, ActionsExecutable, Outputable, Sampleable {
+
+	public final static String ENTITY_PREFIX = "home";
+
+	@Override
+	public String[] getHandledOutputFormats() {
+		return new String[] { Formats.JSON, Formats.XML, Formats.HTML };
+	}
+
+	@Override
+	public Object getSampleEntity() {
+		return "<HTML></HTML>";
+	}
+
+	@Override
+	public String getEntityPrefix() {
+		return ENTITY_PREFIX;
+	}
+
+	@Setter
+	private transient SiteService siteService;
+
+	@Setter
+	private transient ContentHostingService contentService;
+
+	@Setter
+	protected SessionManager sessionManager;
+
+	@Setter
+	private transient DeveloperHelperService developerHelperService;
+
+	private static Logger log = LoggerFactory.getLogger(SiteEntityProvider.class);
+
+	public void init() {
+	}
+
+	@EntityCustomAction(action="migrate",viewKey=EntityView.VIEW_LIST)
+	public Object getEntity(EntityView view, Map<String, Object> params) {
+		String siteId = view.getPathSegment(2);
+		if (siteId == null || "".equals(siteId)) {
+			throw new IllegalArgumentException("No siteId in provided reference");
+		}
+
+		Site site;
+		try {
+			site = siteService.getSite(siteId);
```

Comment:
This probably wants to be a `getSiteVisit(String)` call so that it checks permissions. Technically if it's in the public list of sites then returning then it's ok to return the description without having the ability to visit the site but for the purposes of migration we probably don't need to worry about that.